staging/src/k8s.io/kubectl/pkg/util/rbac/rbac.go

```diff
@@ -42,7 +43,12 @@ func CompactRules(rules []rbacv1.PolicyRule) ([]rbacv1.PolicyRule, error) {
 				if existingRule.Verbs == nil {
 					existingRule.Verbs = []string{}
 				}
-				existingRule.Verbs = append(existingRule.Verbs, rule.Verbs...)
+				uniqueVerbs := sets.NewString(existingRule.Verbs...)
+				for _, v := range rule.Verbs {
```

Comment:
You could just:
```
uniqueVerbs.Insert(rule.Verbs...)
```

Reply:
I believe I have to iterate. I've stared at this for a bit.. maybe I'm not seeing what you're seeing. I'll keep staring : ) I did modify the name from `uniqueVerbs` to `existingVerbs` to make it more clear, though.

Reply:
I was thinking about something like:
```
existingVerbs := sets.NewString(existingRule.Verbs...)
existingVerbs.Insert(rule.Verbs...)
existingRule.Verbs = existingVerbs.List()
```
but your solution is equal to that.